### PR TITLE
Cv 11 statystyki uzytkownikow

### PR DESCRIPTION
**Pull request by Bondpl**

### CODE REVIEW

File: src/main/java/dev/chatverse/backend/controllers/StatisticController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import dev.chatverse.backend.services.UserService;
+
+import java.security.Principal;
+import java.util.List;
+import java.util.Set;
+
+@RequestMapping("/statistic")
```
Comment (bykowskiolaf) [inline src/main/java/dev/chatverse/backend/controllers/StatisticController.java]: consider renaming to plural `statistics`

File: src/main/java/dev/chatverse/backend/controllers/StatisticController.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    private UserService userService;
+    private StatisticService StatisticService;
+
+
+    @GetMapping("/me")
+    public StatisticResponse statistic(Principal principal) {
+
+        String currentPrincipalName = principal.getName();
+
+        return userService.getStatisticResponse(currentPrincipalName);
```
Comment (bykowskiolaf) [inline src/main/java/dev/chatverse/backend/controllers/StatisticController.java]: remove new lines to fit our code style better

File: src/main/java/dev/chatverse/backend/services/StatisticService.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                () -> new UsernameNotFoundException("User not found with email: " + email)
+        );
+    }
+
+    public StatisticResponse getStatisticResponse(String email) {
+        User user = loadUserByEmail(email);
+        return new StatisticResponse(user.getId(), user.getMessageCount(), user.getToxicMessages(), user.getToxicPercentage());
+    }
+
+    public Set<StatisticResponse> getAllUsersStatistic() {
```
Comment (bykowskiolaf) [inline src/main/java/dev/chatverse/backend/services/StatisticService.java]: should be plural `statistics` if Set<> is returned

**Review by bykowskiolaf — CHANGES_REQUESTED** [review]

File: src/main/java/dev/chatverse/backend/controllers/StatisticController.java (hunk 1 above)
Comment (Bondpl) [reply src/main/java/dev/chatverse/backend/controllers/StatisticController.java]: done

File: src/main/java/dev/chatverse/backend/controllers/StatisticController.java (hunk 2 above)
Comment (Bondpl) [reply src/main/java/dev/chatverse/backend/controllers/StatisticController.java]: done

File: src/main/java/dev/chatverse/backend/services/StatisticService.java (hunk 3 above)
Comment (Bondpl) [reply src/main/java/dev/chatverse/backend/services/StatisticService.java]: done

**Review by bykowskiolaf — APPROVED** [review]
Great job! ✨

**Pull request merged**